integrationTests/chainSimulator/bridge/executeAndBurn_test.go

```diff
@@ -0,0 +1,308 @@
+	})
+
+	simulateExecutionAndDeposit(t, mintTokens, depositTokens)
+}
+
+func simulateExecutionAndDeposit(
+	t *testing.T,
+	mintTokens []chainSim.ArgsDepositToken,
+	depositTokens []chainSim.ArgsDepositToken,
+) {
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	whiteListedAddress := "erd1qqqqqqqqqqqqqpgqmzzm05jeav6d5qvna0q2pmcllelkz8xddz3syjszx5"
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:      false,
+		TempDir:                     t.TempDir(),
+		PathToInitialConfig:         defaultPathToInitialConfig,
+		NumOfShards:                 1,
+		GenesisTimestamp:            time.Now().Unix(),
+		RoundDurationInMillis:       uint64(6000),
+		RoundsPerEpoch:              roundsPerEpoch,
+		ApiInterface:                api.NewNoApiInterface(),
+		MinNodesPerShard:            3,
+		MetaChainMinNodes:           3,
+		NumNodesWaitingListMeta:     0,
+		NumNodesWaitingListShard:    0,
+		ConsensusGroupSize:          1,
+		MetaChainConsensusGroupSize: 1,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.GeneralConfig.VirtualMachine.Execution.TransferAndExecuteByUserAddresses = []string{whiteListedAddress}
+			cfg.SystemSCConfig.ESDTSystemSCConfig.BaseIssuingCost = issuePrice
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	nodeHandler := cs.GetNodeHandler(0)
+
+	initialAddress := "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl"
+	initialAddrBytes, err := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode(initialAddress)
+	require.Nil(t, err)
+	err = cs.SetStateMultiple([]*dtos.AddressState{
+		{
+			Address: initialAddress,
+			Balance: "10000000000000000000000",
+		},
+		{
+			Address: "erd1lllllllllllllllllllllllllllllllllllllllllllllllllllsckry7t", // init sys account
+		},
+	})
+	require.Nil(t, err)
+
+	err = cs.GenerateBlocksUntilEpochIsReached(3)
+	require.Nil(t, err)
+
+	wallet := dtos.WalletAddress{Bech32: initialAddress, Bytes: initialAddrBytes}
+	nonce := uint64(0)
+
+	bridgeData := DeployBridgeSetup(t, cs, wallet.Bytes, &nonce, esdtSafeWasmPath, feeMarketWasmPath)
+
+	esdtSafeEncoded, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Encode(bridgeData.ESDTSafeAddress)
+	require.Equal(t, esdtSafeEncoded, whiteListedAddress)
+
+	// We will deposit an array of prefixed tokens from a sovereign chain to the main chain,
+	// expecting these tokens to be minted by the whitelisted ESDT safe sc and transferred to our wallet address.
+	executeBridgeOperation(t, cs, wallet, &nonce, bridgeData.ESDTSafeAddress, mintTokens)
+
+	// Deposit an array of tokens from main chain to sovereign chain,
+	// expecting these tokens to be burned by the whitelisted ESDT safe sc
+	Deposit(t, cs, wallet.Bytes, &nonce, bridgeData.ESDTSafeAddress, depositTokens, wallet.Bytes)
+	mintedTokens := groupTokens(mintTokens)
+	for _, token := range groupTokens(depositTokens) {
```

Comment:
```suggestion
	for _, bridgedOutToken := range groupTokens(depositTokens) {
```

Perhaps after refactor? or burned token?